R/disease_progression.R

```diff
@@ -1,3 +1,79 @@
+#' @title Calculate recovery rates
+#' @description Calculates recovery rates for each individual in the population
+#' for storage in competing hazards object and future resolution
+#'
+#' @param variables the available human variables
+#' @param parameters model parameters
+#' @param recovery_outcome competing hazards object for recovery rates
+#' @noRd
+calculate_recovery_rates <- function(variables, parameters, recovery_outcome, dt_input){
+  
+  # Get correct input for dt depending on antimalarial resistance
+  if(isFALSE(parameters$antimalarial_resistance) & length(dt_input) == 1 & is.numeric(dt_input)){
+    dt_v <- dt_input
+  } else if (isTRUE(parameters$antimalarial_resistance)) {
+    dt_v <- dt_input$get_values(variables$state$get_index_of("Tr"))
+  }
```

Comment:
We probably want to cover all cases and use boolean instead of bitwise operators.

```suggestion
  if(isFALSE(parameters$antimalarial_resistance)){
    stopifnot(length(dt_input) == 1 && is.numeric(dt_input)) # please move this check to pre-simulation time for efficiency
    dt_v <- dt_input
  } else {
    dt_v <- dt_input$get_values(variables$state$get_index_of("Tr"))
  }
```

Reply:
I've made this change.